cmake/setup_GTest.cmake

```diff
@@ -0,0 +1,31 @@
+find_package(GTest)
+
+if (NOT GTest_FOUND)
+    message(STATUS "Configure GTest from github")
+
+    include(FetchContent)
+    FetchContent_Declare(
+      googletest
+      GIT_REPOSITORY https://github.com/google/googletest.git
+      GIT_TAG        v1.13.0
+    )
```

Comment:
After running the tests the other day, I actually think it would better if you inserted the following:

```cmake
    # NOTE: it is idiomatic to use FetchContent with a git hash rather than the
    # name of a tag or branch (since the latter 2 can freely change). If we use
    # the name of a tag/branch, then CMake will query GitHub on every
    # subsequent build to check whether the tag/branch is still the same

    FetchContent_Declare(
      googletest # the url contains the hash for v1.13.0
      URL https://github.com/google/googletest/archive/b796f7d44681514f58a683a3a71ff17c94edb0c1.zip
    )
```

Both what you are currently doing and what I had previously suggested will both clone a git repository into the build directory.  It turns out that this leads to some warnings from the ``rm`` command (at least on macOS) whenever I want to delete the build-directory because the git metadata files are all "protected". (I get a warning like: "are you sure you want to delete this protected file")

What I am now suggesting will download a zip file from GitHub, which will exclude the git metadata and will prevent these warnings.
- It turns out that the [googletest quick-start guide](https://google.github.io/googletest/quickstart-cmake.html#set-up-a-project) also suggests downloading zip files (presumably for the same reasons).
- If you choose to go this route, you will should make a minor tweak to the root `CMakeLists.txt` file. Between lines 24 and 25 (i.e. before the `project(Grackle VERSION ...)` line), you should insert the following snippet:
  ```cmake
  if(POLICY CMP0135)
    cmake_policy(SET CMP0135 NEW)
  endif()
  ```
  This will handle some warnings that show up in newer versions of cmake

If you prefer, I can push a commit that handles this

Reply:
Done.